src/core_plugins/tagcloud/public/tag_cloud.js

```diff
@@ -198,64 +214,77 @@ class TagCloud extends EventEmitter {
 
   };
 
-
   _makeTextSizeMapper() {
-    this._mapSizeToFontSize = D3_SCALING_FUNCTIONS[this._textScale];
-    if (this._words.length === 1) {
-      this._mapSizeToFontSize.range([this._maxFontSize, this._maxFontSize]);
-    } else {
-      this._mapSizeToFontSize.range([this._minFontSize, this._maxFontSize]);
-    }
-
+    const mapSizeToFontSize = D3_SCALING_FUNCTIONS[this._textScale]();
+    const range = this._words.length === 1 ? [this._maxFontSize, this._maxFontSize] : [this._minFontSize, this._maxFontSize];
+    mapSizeToFontSize.range(range);
     if (this._words) {
-      this._mapSizeToFontSize.domain(d3.extent(this._words, getSize));
+      mapSizeToFontSize.domain(d3.extent(this._words, getValue));
     }
+    return mapSizeToFontSize;
   }
 
-  _invalidate(keepLayout) {
+  _makeJob() {
+    return {
+      words: this._words.map(toWordTag)
+    };
+  }
 
+  _invalidate(keepLayout) {
 
     if (!this._words) {
       return;
     }
 
-    if (!this._dirtyPromise) {
-      this._dirtyPromise = new Promise((resolve, reject) => {
-        this._resolve = resolve;
-      });
-    }
-
-    clearTimeout(this._timeoutHandle);
-    this._timeoutHandle = requestAnimationFrame(() => {
-      this._timeoutHandle = null;
+    clearTimeout(this._handle);
+    this._handle = setTimeout(() => {
+      this._handle = null;
       this._updateContainerSize();
-      if (keepLayout) {
-        this._onLayoutEnd();
+      if (keepLayout && this._currentJob && this._queue.length === 0) {
+        this._scheduleLayout({
+          words: this._currentJob.words.map(tag => {
+            return {
+              x: tag.x,
+              y: tag.y,
+              rotate: tag.rotate,
+              size: tag.size,
+              text: tag.text
+            };
+          })
+        });
       } else {
-        this._washWords();
         this._updateLayout();
       }
-    });
+    }, 10);
```

Comment:
I added the time-out so as not to trigger a refresh synchronously. Those 10-milliseconds are a magic number.

In hindsight, I prefer it over the requestAnimationFrame, because the DOM-updates are managed by d3, and requestAnimationFrame is better used for drawing to a canvas anyway.